autotag.go

```diff
@@ -159,6 +159,30 @@ func NewRepo(cfg GitRepoConfig) (*GitRepo, error) {
 		return nil, err
 	}
 
+	if cfg.Branch == "" {
+		branches, err := repo.Branches()
+		if err != nil {
+			return nil, err
+		}
+		for _, b := range branches {
+			if b == "master" {
+				cfg.Branch = "master"
+				break
+			}
```

Comment:
could avoid having to seek through the branches 2 times by setting something like `hasMain`  in this loop. since we will find master or main  guaranteed in this loop. Then the logic on 174 would just be if  hasMain   and removes the 2nd loop through branches.

Reply:
in general I feel like we can remove some nesting, and get more declarative.

Reply:
Also it's quite likely that if main exists And master exists then main should be selected.   IMO we should default to main and find master instead.

Reply:
Previously, I chose master for 100% backwards compatibility.  I agree, though, that main should be preferred going forward.  I'll make that change.